### PR TITLE
New rationale paper.

### PR DESCRIPTION
**Pull request by tclune**

### CODE REVIEW

**Comment by tclune** [comment]
Good point.  Will do.

File: J3-Papers/deferred-polymorphic.txt [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+for the deferred type.  This paper summarizes the rationale by
+generics subgroup for how to support this capability in F202Y.
+
+
+2. Polymorphic deferred use cases
+=================================
+
+Consider a List as a protypical category of container.  A programmer
+may wish to declare a list of reals, a list of integers, or a list of
+some user-define type.  As shown in previous tutorials, the existing
```
Comment (arjenmarkusdeltaresnl) [inline J3-Papers/deferred-polymorphic.txt:23]: user-defined (minute typo)

File: J3-Papers/deferred-polymorphic.txt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    polymorphic.  E.g., CLASS, DEFERRED :: T
+
+
+Subgroup did not like option (1) as it could be relatively difficult
+for the casual programmer to see whether a deferred type is
+polymorphic.  Some obscure usage deep in the source could trigger the
+difference.
+
+Subgroup decided that option (2) would be attractive if related
+intrinsic requirementsn were also to be introduced.    E.g.,
```
Comment (arjenmarkusdeltaresnl) [inline J3-Papers/deferred-polymorphic.txt:69]: requirements - drop the "n"

File: J3-Papers/deferred-polymorphic.txt [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+3. Options for polymorphic deferred
+===================================
+
+Subgroup considered the following options for allowing polymorphic
+deferred objects:
+
+(1) Implicitly interpret the deferred type T as polymorphic if the
+    template references CLASS(T) anywhere in the template.
+
+(2) Introduce _intrinsic_ requirement that indicates that T is
```
Comment (arjenmarkusdeltaresnl) [inline J3-Papers/deferred-polymorphic.txt:52]: Introduce an _intrinsic_ ...

File: J3-Papers/deferred-polymorphic.txt [hunk 4]
```diff
@@ -80,5 +80,17 @@ Our suggested syntax would be:
 
     CLASS, DEFERRED :: T
 
+4. Implications and consequences
+================================
+
+If a template declares a deferred type T as polymorphic, then any
+corresponding actual argument shall be a non-extensible type.   I.e.,
```
Comment (everythingfunctional) [inline J3-Papers/deferred-polymorphic.txt:87]:
```suggestion
corresponding actual argument shall not be a non-extensible type.   I.e.,
```

Comment (tclune) [reply J3-Papers/deferred-polymorphic.txt:87]: Pesky little detail.     I'm wondering though whether "shall be an extensible type" is better than the double negative?

File: J3-Papers/deferred-polymorphic.txt [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+Subgroup therefore prefers the more direct approach with option (3).
+Our suggested syntax would be:
+
+    CLASS, DEFERRED :: T
+
+4. Implications and consequences
+================================
+
+If a template declares a deferred type T as polymorphic, then any
+corresponding actual argument shall be a non-extensible type.   I.e.,
```
Comment (tclune) [inline J3-Papers/deferred-polymorphic.txt:87]:
```suggestion
corresponding actual argument shall be an extensible type.   I.e.,
```

**Review by everythingfunctional — APPROVED** [review]

**Pull request merged**